### DISCUSSION TITLE
Automatic REST API documentation

### DISCUSSION THREAD

**Discussion by gomboc1985** [aws-powertools/powertools-lambda-python#2242] [General]
Hello @heitorlessa, @rubenfonseca and all,

a crucial question please.

Is there any recipe for generating Swagger-like documentation for API defined via APIGatewayRestResolver?
As of now this feature does not seems to be supported out-of-the-box but looks to be a crucial one for any client-facing deliverable. Is there any recommended work-around for achieving such a goal?

Ideally, there should be the equivalent of the automatically-generated swagger documentation page generated by FastAPI (see for example [here](https://fastapi.tiangolo.com/#interactive-api-docs)).

By the way: great job! Love using aws-lambda-powertools fro Python!

**Comment by leandrodamascena** [comment]
Hello @gomboc1985, we already have an [issue](https://github.com/awslabs/aws-lambda-powertools-python/issues/1236) open for this feature request. We don't have an ETA yet to release this, but it could be awesome if you add comments with your need to help us prioritize it.

And it's great to hear that you're really enjoying using Powertools! 🚀

https://github.com/awslabs/aws-lambda-powertools-python/issues/1236

**Reply by gomboc1985** [reply]
Thanks a lot for the quick reply!
